### PR TITLE
Remove dummy layer

### PR DESCRIPTION
**Pull request by cheungdaven**
*Issue #, if available:*

*Description of changes:*

1. overwrite the max length for standalone checkpoints.
2. remove dummy layer to enable gradient_checkpointing 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: multimodal/src/autogluon/multimodal/data/process_ner.py [hunk 1]
```diff
@@ -124,7 +124,9 @@ def process_ner(
         A dictionary containing one NER sample's features and/or labels.
         """
         ret = {}
-
+        # overwrite model_max_length for standalone checkpoints if it's not specified.
+        if self.max_len is not None and self.tokenizer.model_max_length > 10**6:
+            self.tokenizer.model_max_length = self.max_len
```
Comment (zhiqiangdon) [inline multimodal/src/autogluon/multimodal/data/process_ner.py:129]: Why not doing this in `init`?

Comment (cheungdaven) [reply multimodal/src/autogluon/multimodal/data/process_ner.py:129]: It's done in init. But for checkpoints that are saved using the standalone mode, the tokenizer will be overwritten in load.py and the model_max_length is gone so here we rewrite it.

**Review by zhiqiangdon — APPROVED** [review]
LGTM.

File: multimodal/src/autogluon/multimodal/data/process_ner.py (hunk 1 above)
Comment (zhiqiangdon) [reply multimodal/src/autogluon/multimodal/data/process_ner.py:129]: It seems that `save_text_tokenizers` missed the max_len attribute.

Comment (sxjscience) [reply multimodal/src/autogluon/multimodal/data/process_ner.py:129]: Shall we save max_len instead?

Comment (sxjscience) [reply multimodal/src/autogluon/multimodal/data/process_ner.py:129]: Or we can rely on our saved yaml config

**Review by sxjscience — APPROVED** [review]
LGTM

File: multimodal/src/autogluon/multimodal/data/process_ner.py (hunk 1 above)
Comment (cheungdaven) [reply multimodal/src/autogluon/multimodal/data/process_ner.py:129]: I will add a TODO and revise it later.

**Pull request merged**